spec/spec_helper_methods.rb

```diff
@@ -0,0 +1,26 @@
+def os_specific_facts(facts)
```

Comment:
Why don't you define it as a custom fact so you don't need to override it in actual tests? https://github.com/theforeman/puppet-puppet/blob/f053a8fbdabeb328cc0af9b432a0ce6d626ca008/spec/spec_helper.rb#L6-L31 is the one I always use. By using comparisons rather than listing them explicitly, it's more future proof.

Reply:
To be honest, I do not understand why we are using ```service_provider``` if ```camptocamp/puppet-systemd``` provides its own fact ```systemd``` based on the ```service_provider```.

Reply:
Never mind. I just figured it out.